source/common/upstream/cds_api_impl.cc

```diff
@@ -68,6 +73,19 @@ void CdsApiImpl::onConfigUpdate(const ResourceVector& resources, const std::stri
 
   version_info_ = version_info;
   runInitializeCallbackIfAny();
+  throwIfExceptionOccured(exception_msgs);
+}
+
+void CdsApiImpl::throwIfExceptionOccured(const std::list<std::string>& exception_msgs) {
+  if (!exception_msgs.empty()) {
+    std::stringstream sstream;
+    for (auto& msg : exception_msgs) {
+      sstream << msg << "\n";
+    }
+    auto message = sstream.str();
+    message.pop_back();
```

Comment:
This can be replaced with https://github.com/envoyproxy/envoy/blob/master/source/common/common/utility.h#L297